contributor-guide/modules/ROOT/pages/testing/writing-tests.adoc

```diff
@@ -1,3 +1,499 @@
 = Writing Tests
+:navtitle: Writing Tests
 
-*Content under construction.*
\ No newline at end of file
+== Guidelines for Contributors
+
+By following these guidelines and examples, contributors can write effective and maintainable unit tests that ensure the robustness and reliability of their Boost library contributions.
+
+. Tests should be neatly organized into test suites to maintain clarity about which aspect of the library is being tested.
+
+. When testing features that are compiler or platform-specific, use boost:config[] to ensure portability.
+
+. Test names should be descriptive enough to understand the purpose without diving deep into the test logic.
+
+. If a test contains non-trivial logic or checks for edge cases, include comments explaining the rationale.
+
+. Ensure that the tests cover all corner cases, edge conditions, and typical usage scenarios.
+
+. Each test should be independent, not relying on the state or outcome of another test.
+
+. Well-commented unit tests not only describe the "what" but also the "why" behind certain checks, making them invaluable for both current developers and future maintainers. It's crucial for unit tests to be self-explanatory, and while descriptive names play a significant role, comments can further elucidate complex or non-obvious logic.
+
+== Unit Tests
+
+Writing good unit tests is crucial. boost:test[] provides the facilities needed to write unit tests, and boost:config[] can be used to adjust code depending on the platform and compiler features. The following are some examples to help contributors get started:
+
+=== Basic Unit Testing using Boost.Test
+
+To begin, contributors should include the necessary headers and use the right macros.
```

Comment:
This sentence seems weird because it looks goal-oriented but it's not. 

There's a lot that comes before including the headers and macros, like properly compiling and linking the library. That means they will need to read the Boost.Test documentation anyway.

And once this is done without the help of this tutorial, the user already knows about headers and macros because that's trivial in comparison.